scope_ascom.cpp

```diff
@@ -190,11 +196,13 @@ void ScopeASCOM::SetupDialog()
             wxMessageBox(msg, _("Error"), wxOK | wxICON_ERROR);
         }
     }
-    // destroy the COM object now as this reduces the likelhood of getting into a
+    // If we just created the COM object for doing the SetupDialog (device not already connected),
+    // we need to destroy it now as this reduces the likelhood of getting into a
     // state where the user has killed the ASCOM local server instance and PHD2 is
     // holding a reference to the defunct driver instance in the global interface
     // table
-    m_gitEntry.Unregister();
+    if (!prevRegistered)

```

Comment:
let's add a comment explaining why this is conditional:
```suggestion
    // Only un-register from the GIT if we happened to have done the registration here
    // in this function call. Otherwise, if the object was already registered at the time
    // we came into this function, de-registration will be taken care of elsewhere.
    if (!prevRegistered)
```